InterfaceFortran/fortester.f90

```diff
@@ -27,184 +27,1471 @@
+points => tsgGetPoints(gridID)
+allocate(double_2d_b(4,tsgGetNumPoints(gridID)))
+double_2d_b(1,:) = 0.3d0
+double_2d_b(2,:) = points(1,:)    + points(2,:)
+double_2d_b(3,:) = points(1,:)**2 + points(2,:)**2 + points(1,:)*points(2,:)
+double_2d_b(4,:) = points(1,:)**3 + points(2,:)**3 + points(1,:)*(points(2,:)**2)
+call tsgLoadNeededPoints(gridID, double_2d_b)
+call tsgEvaluate(gridID,reshape(pointsb(:,1),(/2/)),double_1d_a)
+call tsgEvaluate(gridID,reshape(pointsb(:,2),(/2/)),double_1d_b)
+call tsgEvaluateBatch(gridID,pointsb,2,double_2d_c)
+do i=1,3
+  if ( norm1d(double_2d_a(i,:)-double_2d_c(i,:)) > 1.d-11 .or. &
+      ( abs(double_2d_a(i,1)-double_1d_a(i)) + abs(double_2d_a(i,2)-double_1d_b(i)) ) > 1.d-11 ) then
+    write(*,*) "Mismatch in tsgEvaluateBatch: case 4, output ", i
+    stop 1
+  endif
+enddo
+do i=4,4
+  if ( ( norm1d(double_2d_a(i,:)-double_2d_c(i,:)) < 1.d-8 ) .or. &
+       ( abs(double_2d_a(i,1)-double_1d_a(i)) + abs(double_2d_a(i,2)-double_1d_b(i)) ) < 1.d-8 ) then
+    write(*,*) "Mismatch in tsgEvaluateBatch: case 4, output ", i
+    stop 1
+  endif
+enddo
+deallocate(double_2d_a, double_2d_b, double_2d_c, double_1d_a, double_1d_b, pointsb, points)
+
+
+allocate(double_2d_a(4,2), double_2d_c(4,2), double_1d_a(4), double_1d_b(4), pointsb(2,2))
+pointsb = reshape( (/ 1./3.d0, 1./3.d0, pi/6.d0, -sqrt(2.d0)/2.d0 /), shape(pointsb) )
+double_2d_a(1,:) = 0.3d0
+double_2d_a(2,:) = pointsb(1,:)    + pointsb(2,:)
+double_2d_a(3,:) = pointsb(1,:)**2 + pointsb(2,:)**2 + pointsb(1,:)*pointsb(2,:)
+double_2d_a(4,:) = pointsb(1,:)**3 + pointsb(2,:)**3 + pointsb(1,:)*(pointsb(2,:)**2)
+call tsgMakeLocalPolynomialGrid(gridID, 2, 4, 3, 3, tsg_localp)
+points => tsgGetPoints(gridID)
+allocate(double_2d_b(4,tsgGetNumPoints(gridID)))
+double_2d_b(1,:) = 0.3d0
+double_2d_b(2,:) = points(1,:)    + points(2,:)
+double_2d_b(3,:) = points(1,:)**2 + points(2,:)**2 + points(1,:)*points(2,:)
+double_2d_b(4,:) = points(1,:)**3 + points(2,:)**3 + points(1,:)*(points(2,:)**2)
+call tsgLoadNeededPoints(gridID, double_2d_b)
+call tsgEvaluate(gridID,reshape(pointsb(:,1),(/2/)),double_1d_a)
+call tsgEvaluate(gridID,reshape(pointsb(:,2),(/2/)),double_1d_b)
+call tsgEvaluateBatch(gridID,pointsb,2,double_2d_c)
+do i=1,4
+  if ( ( norm1d(double_2d_a(i,:)-double_2d_c(i,:)) > 1.d-11 ) .or. &
+       ( abs(double_2d_a(i,1)-double_1d_a(i)) + abs(double_2d_a(i,2)-double_1d_b(i)) ) > 1.d-11 ) then
+    write(*,*) "Mismatch in tsgEvaluateBatch: case 5, output ", i
+    stop 1
+  endif
+enddo
+deallocate(double_2d_a, double_2d_b, double_2d_c, double_1d_a, double_1d_b, pointsb, points)
+
+
+call tsgMakeGlobalGrid(gridID, 2, 1, 22, tsg_iptotal, tsg_chebyshev)
+points => tsgGetPoints(gridID)
+allocate(double_2d_a(1,tsgGetNumPoints(gridID)))
+double_2d_a(1,:) = exp( -points(1,:)**2 - points(2,:)**2 )
+call tsgLoadNeededPoints(gridID, double_2d_a)
+allocate(pointsb(2,1000),double_2d_b(1,1000),double_2d_c(1,1000))
```

Comment:
you are setting `pointsb` here for evaluation, but you are not initializing with random values